samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemAdmin.scala

```diff
@@ -554,14 +573,31 @@ class KafkaSystemAdmin(
     topicMetadata.partitionsMetadata.isEmpty
   }
 
+  /**
+    * @inheritdoc
+    *
+    * Delete records up to (and including) the provided ssp offsets for all system stream partitions specified in the map
+    * This only works with Kafka cluster 0.11 or later. Otherwise it's a no-op.
+    */
+  override def deleteMessages(offsets: util.Map[SystemStreamPartition, String]) {
+    if (!running)
+      throw new SamzaException("KafkaSystemAdmin has not started yet")
+    val nextOffsets = offsets.asScala.toSeq.map { case (systemStreamPartition, offset) =>
+      (new TopicPartition(systemStreamPartition.getStream, systemStreamPartition.getPartition.getPartitionId), offset.toLong + 1)
+    }.toMap
+
+    if (deleteMessagesEnabled)
+      adminClient.deleteRecordsBefore(nextOffsets)
```

Comment:
Is there any perf-overhead with this method since we do it at each commit? If so, would it be possible to invoke this only if `deleteRecordsBefore` supported on the broker? You can make this determination once during startup.